src/Subscriber/SlidingPaginationSubscriber.php

```diff
@@ -41,12 +41,12 @@ public function pagination(PaginationEvent $event): void
         // default sort field and order
         $eventOptions = $event->options;
 
-        if (isset($eventOptions['defaultSortFieldName']) && !isset($this->params[$eventOptions['sortFieldParameterName']])) {
-            $this->params[$eventOptions['sortFieldParameterName']] = $eventOptions['defaultSortFieldName'];
+        if (isset($eventOptions['defaultSortFieldName']) && isset($this->params[$eventOptions['sortFieldParameterName']]) && $eventOptions['defaultSortFieldName'] === $this->params[$eventOptions['sortFieldParameterName']]) {
```

Comment:
I don't get the ratio of this change: you're removing an old condition, so you're changing a behavior instead of adding a new one.

Reply:
i must create option in place like this?
```
// src/Repository/AnimalRepository.php
public function findAnimalsPaginated(array $params, int $page, int $limit = 10): PaginationInterface
    {
        $query = $this->createQueryBuilder('a')
            ->andWhere('a.status = :status')
            ->setParameter('status', Animal::STATUS_ACTIVE);

        $paginateOptions = [
            PaginatorInterface::DEFAULT_SORT_FIELD_NAME => 'a.published_at',
            PaginatorInterface::DEFAULT_SORT_DIRECTION => 'desc',
            'hereNewOption' => true,
        ];
        return $this->paginator->paginate($query->getQuery(), $page, $limit, $paginateOptions);
    }
```
and only if true add new behavior without change old, right?